In [1]:
import pandas as pd
import xgboost as xgb
import prophet

print("All libraries imported successfully!")

/Users/rehan/Documents/weather-trend-forecasting/venv/lib/python3.13/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


All libraries imported successfully!


In [3]:
# Load the dataset (using ../ since the notebook is in the notebooks folder)
df = pd.read_csv('../data/GlobalWeatherRepository.csv')

# Basic sanity checks
print(f"Dataset Shape: {df.shape}")
print("\n--- Data Types ---")
print(df.dtypes.head(10)) # Just looking at the first 10 for now
print("\n--- Missing Values Count (Top 10) ---")
print(df.isnull().sum().sort_values(ascending=False).head(10))

Dataset Shape: (145212, 41)

--- Data Types ---
country                       str
location_name                 str
latitude                  float64
longitude                 float64
timezone                      str
last_updated_epoch          int64
last_updated                  str
temperature_celsius       float64
temperature_fahrenheit    float64
condition_text                str
dtype: object

--- Missing Values Count (Top 10) ---
country                         0
feels_like_fahrenheit           0
visibility_miles                0
uv_index                        0
gust_mph                        0
gust_kph                        0
air_quality_Carbon_Monoxide     0
air_quality_Ozone               0
air_quality_Nitrogen_dioxide    0
air_quality_Sulphur_dioxide     0
dtype: int64


In [4]:
# Convert to datetime object
df['last_updated'] = pd.to_datetime(df['last_updated'])

# Extract useful time features
df['date'] = df['last_updated'].dt.date
df['year'] = df['last_updated'].dt.year
df['month'] = df['last_updated'].dt.month
df['hour'] = df['last_updated'].dt.hour

# Check the date range to see exactly how much historical data we have
print(f"Data starts: {df['last_updated'].min()}")
print(f"Data ends: {df['last_updated'].max()}")

Data starts: 2024-05-16 01:45:00
Data ends: 2026-06-03 19:00:00


### Initial Data Assessment
* **Coverage:** The dataset contains [145212] rows and [41] columns.
* **Date Range:** The data spans from [2024-05-16] to [2026-06-03]. 
* **Missing Data:** No missing data across all 41 columns.